draft-ietf-quic-transport.md

```diff
@@ -2978,6 +3062,79 @@ advised to bundle as few streams as necessary in outgoing packets without losing
 transmission efficiency to underfilled packets.
 
 
+## CRYPTO_HS Frame {#frame-crypto}
+
+The CRYPTO_HS frame (type=0x18) is used to transmit cryptographic handshake
+messages. It can be sent in all packet types. The CRYPTO_HS frame offers the
+cryptographic protocol an in-order stream of bytes.
+
+
+A CRYPTO_HS frame is shown below.
+
+~~~
+ 0                   1                   2                   3
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                         [Offset (i)]                        ...
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                         [Length (i)]                        ...
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                        Crypto Data (*)                      ...
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+~~~
+{: #crypto-format title="CRYPTO_HS Frame Format"}
+
+The CRYPTO_HS frame contains the following fields:
+
+Offset:
+
+: A variable-length integer specifying the byte offset in the stream for the
+  data in this CRYPTO_HS frame.
+
+Length:
+
+: A variable-length integer specifying the length of the Crypto Data field in
+  this CRYPTO_HS frame.
+
+Crypto Data:
+
+: The cryptographic message data.
+
+The CRYPTO_HS stream starts off at an offset of 0.  At each packet
+number space, the sender resets the CRYPTO_HS stream to an offset of
+0.  This implies that each encryption level is treated as a separate
+CRYPTO_HS stream of data.
+
+While CRYPTO_HS the frame looks a lot like a STREAM frames, it should
```

Comment:
Awkward wording, and I find the paragraph a bit confusing anyway.  The first two sentences are implied by the compare/contrast in the latter part.  Maybe, "Unlike STREAM frames, which include a Stream ID indicating to which stream the data belongs, the CRYPTO_HS frame carries data for a single stream per packet number space.  The stream does not have an explicit end, so CRYPTO_HS frames do not have a FIN bit."

Reply:
Done.